src/physics/em/detail/Rayleigh.hh

```diff
@@ -9,30 +9,124 @@
 
 #include "base/Macros.hh"
 #include "base/Types.hh"
+#include "base/Collection.hh"
+#include "physics/base/Types.hh"
 
 namespace celeritas
 {
+struct ModelInteractPointers;
+
 namespace detail
 {
+//---------------------------------------------------------------------------//
+/*!
+ * Rayleigh angular parameters (form factor) for sampling the angular
+ * distribution of coherently scattered photon
+ */
+constexpr unsigned int rayleigh_num_parameters = 9;
+constexpr unsigned int rayleigh_num_elements   = 100;
+
+struct RayleighData
+{
+    static real_type angular_parameters[rayleigh_num_parameters]
+                                       [rayleigh_num_elements];
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Storage for Rayleigh angular parameters
+ */
+struct RayleighElementData
+{
+    Real3 n;
+    Real3 b;
+    Real3 x;
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Storage for Rayleigh angular parameters
+ */
+template<Ownership W, MemSpace M>
+struct RayleighParameters
+{
+    //    using IntId = celeritas::ItemId<int>;
+    using UintId = celeritas::ItemId<unsigned int>;
+
+    template<class T>
+    using Items = celeritas::Collection<T, W, M, UintId>;
+
+    Items<Real3> data_n;
+    Items<Real3> data_b;
+    Items<Real3> data_x;
+
+    explicit CELER_FUNCTION operator bool() const
+    {
+        return !data_n.empty() & !data_b.empty() & !data_x.empty();
+    }
+
+    //! Assign from another set of parameters
+    template<Ownership W2, MemSpace M2>
+    RayleighParameters& operator=(const RayleighParameters<W2, M2>& other)
+    {
+        CELER_EXPECT(other);
+        data_n = other.data_n;
+        data_b = other.data_b;
+        data_x = other.data_x;
+
+        return *this;
+    }
+};
+
 //---------------------------------------------------------------------------//
 /*!
  * Device data for creating an interactor.
  */
-struct RayleighInteractorPointers
+template<Ownership W, MemSpace M>
+struct RayleighPointers
```

Comment:
I think the key is having some phrase that says "this is templated on `<Ownserhip, Memspace>` and has operator= and bool" -- which generally implies having collections, though of course doesn't have to. Something that's safe

Because the earlier Interactor work didn't have `Params` in their name we didn't have it here, but all the interactor data is effectively `Params` -- though I think for Livermore we're also passing in a shared stack allocator 🤔

Reply:
In this case, the content (of RayleighPointer) is constant variables (model_id and gamma_id) and constant parameters (collection).  For now, I will use Group and rename it accordingly once we adopt something better later as other physics interactors will follow similar naming conventions.